README.md

```diff
@@ -0,0 +1,168 @@
+  controlled.
+* Teams can share and collaborate on application deployments described using
+  Puppet's DSL.
+* Users can take advantage of the composition aspects of the Puppet DSL to
+  create reusable and extendable abstraction layers on top of multi-node
+  deployment descriptions.
+* Allows Puppet to support ongoing management of application stacks created in
+  GCE.
+
+## Setup
+In order to use this module, you will need to
+[signup](https://developers.google.com/compute/docs/signup)
+ for a Google Cloud Platform account and enable Google Compute Engine.
+
+The instance to be used as the master must have read/write enabled for compute engine. Add the `--service_account_scopes=compute-rw` flag to the gcutil command when creating the instance or select 'Read Write' under the Compute Project Access when creating the instance through the Developers Console (you may neet to 'show advanced options' to see this field).
+It is not possible to give an existing instance read/write for compute, so be careful to complete this step correctly.
+
+Follow the [documentation](http://docs.puppetlabs.com/guides/install_puppet/install_debian_ubuntu.html) to install Puppet on the node.
+
+For quick testing on Debian Wheezy, try the following:
+
+1. `wget https://apt.puppetlabs.com/puppetlabs-release-wheezy.deb`
+1. `sudo dpkg -i puppetlabs-release-wheezy.deb`
+1. `sudo apt-get update && sudo apt-get upgrade -y`
+1. `sudo apt-get install puppetmaster -y`
+
+For [Fog](fog.io/compute/) on Debian Wheezy install build-essential and ruby-dev with:
+
+1. `sudo apt-get build-essential ruby-dev -y`
+
+Install ruby gems `fog` and `google-api-client` with:
+
+1. `sudo gem install google-api-client --no-rdoc --no-ri`
+1. `sudo gem install fog --no-rdoc --no-ri`
+
+In order to use this repo as a Puppet module, create a symlink to Puppet's
+`modules` directory to your local copy of this git repo:
+
+1. `cd ~`
+1. `git clone https://github.com/GoogleCloudPlatform/gce_compute_fog gce_compute`
+1. `sudo ln -s ~/gce_compute /etc/puppet/modules/gce_compute`
+
+## Usage
+
+### Setting up authorization
+
+#### Auth with temporary tokens (recommended)
+
+This method requires a master with compute read/write permissions and will
+only work if your master is running in Google Compute Engine.  This auth
+method will retrieve an `authorization token` from GCE's internal metadata
+service.
+
+Create type called `gce_auth` with the name as the project id in your manifest. Your resource should look something like:
+```
+gce_auth { 'project-id':
+}
+```
+
+#### Auth using client email and key file
+
+If your master resides outside of Compute Engine, you will need to use a
+Service Account.
+
+1. Create a [GCE service account](https://developers.google.com/compute/docs/faq#howdoicreate). 
+2. Push the key file downloaded from step one to your master VM.
+3. Create a `gce_auth` resource in your manifest file that inculdes the client email and the absolute path to the key file. 
+```
```

Comment:
It's useful to provide a hint that this is puppet code, which on GitHub at least triggers syntax highlighting. Use ```puppet on the first line.